build.gradle

```diff
@@ -65,22 +62,20 @@ license {
 }
 
 wrapper {
-    gradleVersion = '6.8'
+    gradleVersion = '7.0.2'
```

Comment:
From what the image of the conversation in the comment of that PR tells me, it seems that the issue lies with the plugin's use of dynamic Groovy. If I understand it correctly, when in a class with an `extension` field and some code from within a `project.with` closure tries to resolve `extension`, it resolves `project.extension` and not `extension`.

ForgeGradle 5 should be fine with Gradle 7.1 on that basis alone (barring some deprecation warnings because they deprecated `Project#getConvention` and we cannot move away due to `JavaPluginConvention` missing methods in its extension counterpart, `JavaPluginExtension`).

However, I do think that in any case, we should consider moving to Gradle 7.1 since we are currently within the breaking changes window. But that would need confirmation from Lex if we are to move to Gradle 7.1.

Reply:
Jamierocks has not yet pushed the fixed version of Licenser to the gradle plugin portal, meaning we cannot move to 7.1 at this time.

Reply:
Seems they just [did](https://github.com/CadixDev/licenser/commit/6e73b6b63a93c7a2402891c757502bf33f6748ad) so, so once it finishes building this can probably be done.

Reply:
Bumpbed minimum Gradle version to 7.1, and updated the Licenser plugin to 6.0.1. Also went through and removed uses of deprecated code, notably `JavaPluginConvention` (used for getting `SourceSet`s) in favor of `JavaPluginExtension`.

This hasn't been tested yet, would appreciate testing on Gradle 7.1 if it works (and on Gradle 7.0.x to see if the Gradle version env. check works).